Pat/Network/UIImageView+FetchImage.swift

```diff
@@ -3,10 +3,10 @@ import UIKit
 extension UIImageView {
     func fetchImage(from imageURL: URL?,
                     _ completion: @escaping ((Result<UIImage, Error>) -> Void)) -> URLSessionDataTask? {
-
+        let imageCache = ObjectCache.shared
         var dataTask: URLSessionDataTask?
         guard let imageURL = imageURL else { return nil }
-        if let cachedImage = ImageCache.shared.imageCache?[imageURL] {
+        if let cachedImage = imageCache.get(key: imageURL) as? UIImage {
             completion(.success(cachedImage))
         } else {
             let urlSession = URLSession.shared
```

Comment:
faltou colocar a imagem no cache no success aqui

Reply:
To fazendo isso lá na própria cell. Faz muita diferença?

Reply:
Melhor deixar tudo junto, né, já que nesse eu to verificando se tem imagem no cache

Reply:
A ideia de encapsular as coisas é justamente deixar tudo num lugar controlado, assim vc vai saber onde mexer quando precisar. A célula não precisa saber que existe um cache, se vc fizer daqui ela fica protegida dessa complexidade